File: Java/rest_service/src/main/java/restservice/rest_service/controller/GamesController.java
package restservice.rest_service.controller;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.server.ResponseStatusException;
import restservice.rest_service.models.AllGames;
import restservice.rest_service.models.Games;

import java.util.List;

@RestController
@RequestMapping("/geek")
public class GamesController {
    private static final Logger logger = LoggerFactory.getLogger(GamesController.class);

    @GetMapping("/games")
    public ResponseEntity<List<Games>> getAll() {
        logger.info("Buscando lista de vídeo games.");
        return ResponseEntity.ok(AllGames.listGames);
    }

    @GetMapping("/games/owner/{owner}")
    public ResponseEntity<List<Games>> getGame(@PathVariable String owner) {
        logger.info("Buscando vídeo games do mesmo proprietário.");
        var games = AllGames.listGames.stream()
                .filter(x -> x.getOwner().equals(owner))
                .toList();

        if (games.isEmpty()) {
            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Vídeo game não encontrado na lista para esse proprietário.");
        } else {
            return ResponseEntity.ok(games);
        }
    }

    @PostMapping("/games")
    public ResponseEntity<List<Games>> postGame(@RequestBody Games game) {
        logger.info("Acrescentando um vídeo game.");
        AllGames.listGames.add(game);
        return ResponseEntity.ok(AllGames.listGames);
    }

    @DeleteMapping("/games/{name}")
    public ResponseEntity<List<Games>> postGame(@PathVariable String name) {
        logger.info("Removendo um vídeo game.");
        var game = AllGames.listGames.stream()
                .filter(x -> x.getName().equals(name))
                .findFirst();
        if (game.isEmpty()) {
            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Vídeo game não encontrado na lista com esse nome.");
        } else {
            AllGames.listGames.remove(game.get());
            return ResponseEntity.ok(AllGames.listGames);
        }
    }
}

File: C#/rest-service/Controllers/GamesController.cs
using Microsoft.AspNetCore.Mvc;

namespace rest_service.Controllers;

[ApiController]
[Route("geek")]
public class GamesController : ControllerBase
{
    private readonly ILogger<GamesController> _logger;

    public GamesController(ILogger<GamesController> logger)
    {
        _logger = logger;
    }

    [HttpGet("games")]
    public IActionResult GetAll()
    {
        _logger.LogInformation("Buscou lista de vídeo games.");
        return Ok(AllGames.listGames);
    }

    [HttpGet("games/owner/{owner}")]
    public IActionResult GetGame(string owner)
    {
        _logger.LogInformation("Buscou lista de vídeo games.");
        var games = AllGames.listGames.Where(x => x.Owner == owner).ToList();
        if (games == null) {
            return NotFound("Vídeo game não encontrado na lista para essa empresa.");
        } else {
            return Ok(games);
        }
    }

    [HttpPost("games")]
    public IActionResult PostGame(Games games)
    {
        _logger.LogInformation("Acrescentando um vídeo games.");
        AllGames.listGames.Add(games);
        return Created("/", AllGames.listGames);
    }

    [HttpDelete("games/{name}")]
    public IActionResult DeleteGame(string name)
    {
        _logger.LogInformation("Removendo um vídeo games.");
        var game = AllGames.listGames.FirstOrDefault(x => x.Name == name);
        if (game == null) {
            return NotFound("Vídeo game não encontrado na lista com esse nome.");
        } else {
            AllGames.listGames.Remove(game);
            return Ok(AllGames.listGames);
        }
    }
}
